Userland/Libraries/LibVideo/VP9/Parser.cpp

```diff
@@ -19,15 +18,18 @@
 #    pragma GCC optimize("O3")
 #endif
 
-// Beware, threading is unstable in Serenity with smp=on, and performs worse than with it off.
-#define VP9_TILE_THREADING
-
 namespace Video::VP9 {
 
 #define TRY_READ(expression) DECODER_TRY(DecoderErrorCategory::Corrupted, expression)
 
 Parser::Parser(Decoder& decoder)
     : m_decoder(decoder)
+#ifdef VP9_TILE_THREADING
+    , m_thread_pool([this](Work work) {
+        if (auto result = work.decode_tile_column(work.tile_column); result.is_error() && m_process_has_error.exchange(true, AK::MemoryOrder::memory_order_acq_rel))
+            m_process_error = result.release_error();
```

Comment:
I think perhaps I made a bad call to say we should try to get the "first" error, since it requires an extra field, only to get an error that is non-deterministic given that threads can run at different rates at different times.

Not really a big deal, but removing that logic might make this look even cleaner than you already made it. :^)

Reply:
It doesn't make much difference tbh, we need a separate atomic bool to signal "someone won the race", the exact semantics (first or just a random one) only depends on the memory order.
Unless I'm missing an obvious solution to atomically assigning to this variable without a separate atomic object?

Reply:
It doesn't make much difference tbh, we need a separate atomic bool to signal "someone won the race", the exact semantics (first or just a random one) only depends on the memory order.
Unless I'm missing an obvious solution to atomically assigning to this variable without a separate atomic object?

Reply:
I suppose that's true, I guess I wasn't thinking about how writing to the error field would be multiple write operations.